### ISSUE TITLE
Incorrect warnings about GUI RPC requests from IPv4 addresses in 2/8 subnet

### ISSUE THREAD

**Issue by rlpm** [BOINC/boinc#4111]
Upon reimaging one of my RasPis & restarting BOINC, I saw a warning about a connection attempt from a non-allowed IPv4 address in subnet 2/8, e.g.:
```
Nov 25 12:15:48 folder-pi-0 boinc[272]: 25-Nov-2020 12:15:48 [---] GUI RPC request from non-allowed address 2.0.133.136
```
I was very concerned that somehow an attacker had broken through my firewalls, NATs, etc.
Running `tcpdump` for 30 minutes showed nothing from that subnet, or to the BOINC listener. But to be secure, I configured `iptables` on this pi to block & log connection attempts from all IPv4 addresses except one specific node on my internal network. (All my nodes have addresses in 10.0.0/24)
After that, BOINC on this pi logged another non-allowed connection error, again from an address in 2/8. But thankfully I was still running `tcpdump` waiting for it to happen. It turns out this log message coincided with a test I had run from another host.
From that, I determined there is a bug in BOINC's construction of that log message, so it outputs `2.0.x.y` instead of the actual source address. The bug is here:
https://github.com/BOINC/boinc/blob/0c63af1d7b26b27fe5f9f39090a163f19ce77e7f/client/gui_rpc_server.cpp#L394

I'm reasonably sure this should be something like:
```
inet_ntop(s.ss_family, (s.ss_family == AF_INET ? &((sockaddr_in)s).sin_addr : &((sockaddr_in6)s).sin6_addr), buf, 256);
```

Steps to reproduce:
1. Ensure no external hosts are allowed to connect to the Remote GUI RPC (nothing in `allow_remote_gui_rpc` in `/var/lib/boinc/cc_config.xml`)
2. Restart the service (this step is necessary empirically, perhaps the message is not logged on repeated attempts?)
3. From another host on the same network, connect to the Remote GUI RPC , e.g. `echo | nc -v folder-pi-0.local 31416` (replacing the hostname as necessary)
4. Inspect `/var/log/syslog` to find the warning message stating the connection attempt was from 2/8 but actually wasn't.

**Comment by RichardHaselgrove** [comment]
Please take a look at the most recent (of several) attempts to fix this known bug: issue #3340, PR #3342.

If somebody could confirm a working diagnosis and correction, and code it up as a valid PR, it would be appreciated.

**Comment by rlpm** [comment]
Yes, this is a duplicate of issue #3340. Sorry I didn't find it on a cursory look. This should be fixed by #4113, which is a more complete fix than #3342.
